### PR TITLE
[runtime-security] load rules fails gracefully and sends errors as part of the rule event

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

load rules fails gracefully and sends errors as part of the rule event.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Adding rules with syntax error, duplicate IDs, or unknown fields shouldn't avoid to start system-probe and should generate an event with the related errors.

### CODE REVIEW

**Review by guedou — APPROVED** [review]

File: releasenotes/notes/runtime-security-graceful-rule-load-655a6018e6e2d899.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Runtime security module doesn't stop on first policies file
+    load error and now send a event with a report of the load.
```
Comment (guedou) [inline releasenotes/notes/runtime-security-graceful-rule-load-655a6018e6e2d899.yaml:12]: ni: s/a event/an event/

File: pkg/security/secl/generators/accessors/accessors.go [hunk 2]
```diff
@@ -410,6 +415,16 @@ func (m *Model) GetIterator(field eval.Field) (eval.Iterator, error) {
 	return nil, &eval.ErrIteratorNotSupported{Field: field}
 }
 
+func (m *Model) GetEventTypes() []eval.EventType {
+	return []eval.EventType{
+		{{range $Name, $Exists := .EventTypes}}
+			{{if ne $Name "*"}}
+			eval.EventType("{{$Name}}"),
+			{{end}}
```
Comment (lebauce) [inline pkg/security/secl/generators/accessors/accessors.go:423]: {{- if
and
{{ end -}}

File: pkg/config/config.go [hunk 3]
```diff
@@ -821,6 +821,8 @@ func InitConfig(config Config) {
 	config.BindEnvAndSetDefault("runtime_security_config.pid_cache_size", 10000)
 	config.BindEnvAndSetDefault("runtime_security_config.cookie_cache_size", 100)
 	config.BindEnvAndSetDefault("runtime_security_config.agent_monitoring_events", true)
+	config.BindEnvAndSetDefault("runtime_security_config.runtime_enabled", false)
```
Comment (lebauce) [inline pkg/config/config.go:824]: How do we ensure that, after the upgrade of the agent, all the rules won't be ignored ?

Comment (safchain) [reply pkg/config/config.go:824]: I'm going to change the whole logic, it was just to introduce the parameters, but per internal discussion we need to handle them differently

File: pkg/security/module/module.go [hunk 4]
```diff
@@ -237,7 +262,15 @@ func (m *Module) statsMonitor(ctx context.Context) {
 			if err := m.apiServer.SendStats(); err != nil {
 				log.Debug(err)
 			}
-		case <-ctx.Done():
+		case <-heartbeatTicker.C:
+			tags := []string{fmt.Sprintf("version:%s", version.AgentVersion)}
+			if m.config.RuntimeEnabled {
+				_ = m.statsdClient.Gauge("datadog.security_agent.runtime.running", 1, tags, 1)
```
Comment (Gui774ume) [inline pkg/security/module/module.go:268]: Could you maybe pu all the metrics definition in `metrics.go` ?

File: pkg/security/module/module.go [hunk 5]
```diff
@@ -237,7 +262,15 @@ func (m *Module) statsMonitor(ctx context.Context) {
...
+			}
+			if m.config.FIMEnabled {
+				_ = m.statsdClient.Gauge("datadog.security_agent.fim.running", 1, tags, 1)
```
Comment (Gui774ume) [inline pkg/security/module/module.go:271]: Same here :)

**Review by Gui774ume — CHANGES_REQUESTED** [review]
lgtm

**Review by Gui774ume — APPROVED** [review]
lgtm

**Pull request merged**